### PR TITLE
Gradient descent optimizer

### PR DESCRIPTION
**Pull request by araza6**
Since our main users would be people with a background in physics, I am trying to make it worry about writing the optimization routine as little as possible. For that purpose, we have a `train` method called on an object say `opt` such that `opt.train()` essentially gives them good enough parameters of the cost function that they intend to tune (of course, they can change the hyper-parameters in the call to `train` to suit their needs as well.) If the user wants to write their own training loops, they can simply initialize an `opt = SGD()` object and do `opt.step()` in each step of training.

### CODE REVIEW

**Comment by nathanshammah** [comment]
@araza6 what's the status of this PR? It's been idle for a month now. Is something discussed with @quantshah ?

**Comment by araza6** [comment]
We discussed it in the last call. At the moment, we can go by without custom optimizers as JAX provides them already. So it's a bit low-priority for now

**Comment by nathanshammah** [comment]
Can you then please remove it from the first milestone? Can consider closing it (and reopen it), to help focus on high-priority outstanding issues.

**Comment by araza6** [comment]
I will remove it from the first milestone, and add it to third then

**Comment by araza6** [comment]
For now, we can seamlessly use JAX optimizers, closing this for now.
